src/config.ts

```diff
@@ -109,74 +111,107 @@ export class SimbaConfig {
     }
 
     /**
-     * for env vars, we search, in this order:
-     * local project directory for:
-     *  - .simbachain.env
-     *  - simbachain.env
-     *  - .env
+     * this method only gets called once, when a process first tries to retrieve an env var
+     * if SimbaConfig.envVars's values is zero length, then we call this method
      * 
-     * then we look for a SIMBA_HOME env var at the system level,
-     * and within that directory, we search for:
-     *  - .simbachain.env
-     *  - simbachain.env
-     *  - .env
-     * 
-     * @param envVarKey key of environment variable we want to retrieve value
+     * The code is a bit convoluted, so here's the process:
+     * 1. iterate through file names of (.simbachain.env, simbachain.env, .env) in our project root
+     * 2. we then loop through each of our SimbaEnvVarKeys until we find all of them, or
+     * we have finished going through all files (not all env vars are necessary to be set)
+     * 3. we then run through 1-4 again, but we use SIMBA_HOME instead of project root
+     * 4. set as SimbaConfig.envVars once finished
+     * @returns {Promise<Record<any, any>>}
      */
-     public static retrieveEnvVar(envVarKey: SimbaEnvVarKeys): string {
-        if (envVarKey === SimbaEnvVarKeys.SIMBA_AUTH_ENDPOINT) {
-            return DEFAULT_AUTH_ENDPOINT;
-        }
-        
-        // if we've already configured process.env:
-        if (SimbaConfig.simbaEnvVarFileConfigured) {
-            const val = process.env[envVarKey];
-            if (val) {
-                return val;
-            } else {
-                // we don't want to panic if we're just looking for log level - user shouldn't have to set that
-                if (envVarKey === SimbaEnvVarKeys.SIMBATS_LOG_LEVEL) {
-                    return LogLevel.INFO;
-                }
-                const message = `no value found for environment variable ${envVarKey}`;
-                console.error(`SimbaConfig.retrieveEnvVar :: SIMBA : EXIT : ${message}`);
-                throw(message);
-            }
-        }
+    public static setEnvVars(): Record<any, any> {
+        const foundKeys: Array<any> = [];
+        // the following shouldn't need to be changed
+        // has to do with whether authendpoint should be configured, but
+        // for now this is fine
+        foundKeys.push(SimbaEnvVarKeys.SIMBA_AUTH_ENDPOINT);
+        SimbaConfig.envVars[SimbaEnvVarKeys.SIMBA_AUTH_ENDPOINT] = DEFAULT_AUTH_ENDPOINT;
 
-        // first check local directory
+        // first iterate through local project
+            // through each file name
+                // if we have found all our keys, return our object
         for (let i = 0; i < simbaEnvFilesArray.length; i++) {
```

Comment:
Maybe make this a method add call it for local project and simba home? Or a loop? To remove the duplicated code...